pkg/proc/native/proc_darwin.go

```diff
@@ -25,6 +25,7 @@ type OSProcessDetails struct {
 	exceptionPort    C.mach_port_t // mach port for receiving mach exceptions.
 	notificationPort C.mach_port_t // mach port for dead name notification (process exit).
 	initialized      bool
+	halt             bool
```

Comment:
I like the approach for linux where this is no longer state set but rather passed into `trapWaitInternal`. What are your thoughts on doing something similar here?

Reply:
Ideally, yes. In practice I don't want to spend time figuring out how to do it, since we are not using it.